controllers/argocd/route.go

```diff
@@ -366,3 +381,35 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 	}
 	return r.Client.Update(context.TODO(), route)
 }
+
+func shortenHostname(hostname string) (string, error) {
```

Comment:
Should make sure this function handles the case where hostname is empty, and add a test for this case.

Reply:
I have added this in the function https://github.com/argoproj-labs/argocd-operator/pull/1129/files#diff-34c9d2f4dd86a5067f505bc888a3169f2ff71cca7c59bf83e9f083c91d57b1d6R386, please let me know whether I am checking empty hostname in right way and please correct me for unit test as well for this